File: subArrSum.java
import java.util.HashMap;

// Time Complexity : O(n)
// Space Complexity : O(n)
// Did this code successfully run on Leetcode : Yes
// Any problem you faced while coding this : None.

class subArrSum {
    public int subarraySum(int[] nums, int k) {
        int count = 0;
        int sum   = 0;
        HashMap<Integer, Integer> hashMap = new HashMap<>();
        hashMap.put(0,1);

        for(int i = 0; i < nums.length; i++) {
            sum += nums[i];
            
            if(hashMap.containsKey(sum - k)) {
                count += hashMap.get(sum - k);
            }
            
            hashMap.put(sum, hashMap.getOrDefault(sum, 0) + 1);

        }
        return count;
    }
}

File: SubArrSum.py
'''
time complexity: O(n)
space complexity: O(n)
'''
class Solution(object):
    def subarraySum(self, nums, k):
        """
        :type nums: List[int]
        :type k: int
        :rtype: int
        """
        count_dict= {}
        count_dict[0]=1
        count=0
        sums=0
        for i in range(0,len(nums)):
            sums+=nums[i]
            if(sums-k in count_dict):
               count+=count_dict[sums-k]
            if(sums not in count_dict):
                count_dict[sums] = 1
            else:
                count_dict[sums]=count_dict[sums]+1
        return count